tests/tcp.rs

```diff
@@ -830,6 +832,38 @@ fn loopback_to_localhost_v4() -> Result {
     run_localhost_test(IpVersion::V4, bind_addr, connect_addr)
 }
 
+#[test]
+fn loopback_wildcard_public_v4() -> Result {
+    let bind_addr = SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 1234);
+    let connect_addr = SocketAddr::from((Ipv4Addr::new(192, 168, 0, 1), 1234));
+    run_localhost_test(IpVersion::V4, bind_addr, connect_addr)
```

Comment:
Does this verify the stream's addrs are set the same as what tokio::net types return? These need to behave the same way.

Reply:
`run_localhost_test` asserts:

```
assert_eq!(socket.local_addr()?.ip(), connect_addr.ip());
assert_eq!(socket.peer_addr()?.ip(), connect_addr.ip());
```

on both server and client.